### PR TITLE
[DOCS] Add dev mode next step

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
I drew from a few places for this, but as it was for a getting started section, I wasn't sure how deep to go. I worry that the list that mentions what dev mode does may make no sense to someone who has just started using Geth.

I also wonder as this is quite short, wether to just add it to the new getting started guide instead.

PS. This PR builds on https://github.com/ethereum/go-ethereum/pull/20088 so that needs to be merged first.

### CODE REVIEW

File: docs/_getting-started/dev-mode.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+Geth has a development mode which sets up a single node Ethereum test network with a number of options optimized for developing on local machines. You enable it with the `--dev` argument.
+
+Starting geth in dev mode does the following:
+
+-   Initializes the data directory with a testing genesis block
+-   Sets max peers to 0
+-   Turns off discovery by other nodes
+-   Sets the gas price to 0
+-   Uses the ethash consensus engine with clique headers to allows blocks to be mined as fast as your CPU and memory can handle
```
Comment (holiman) [inline docs/_getting-started/dev-mode.md]: It's a separate consensus engine, 
```suggestion
-   Uses the Clique PoA consensus engine with which allows blocks to be mined as-needed without excessive CPU and memory consumption
```

File: docs/_getting-started/dev-mode.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+-   Prevents the consensus (proof of work) difficulty from ever increasing
```
Comment (holiman) [inline docs/_getting-started/dev-mode.md]: Nah, strike this. It doesn't use PoW, but there is a notion of `difficulty` which is used by Clique to figure out which chain is canonical, but that's a bit too much detail probably

File: docs/_getting-started/dev-mode.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+-   Permits 0 block times in the clique consensus engine and forbid empty blocks for 0 periods
```
Comment (holiman) [inline docs/_getting-started/dev-mode.md]: 0 block times - not sure what that means. It's possible to set `dev.period` to have e.g. one block every second, but by default it mines if there are any transactions waiting to be included.

File: docs/_getting-started/dev-mode.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+And check the balance of the account:
+
+```shell
+> eth.getBalance(eth.accounts[1])
+```
+
+If you want to test your dapps with a realistic block time use the `--dev.period` option when you start dev mode:
+
+```shell
+geth --dev --dev.period 100 console
```
Comment (holiman) [inline docs/_getting-started/dev-mode.md]:
```suggestion
geth --dev --dev.period 14 console
```
14 is more realistic than 100

File: docs/_getting-started/dev-mode.md (hunk 2 above)
Comment (ChrisChinchilla) [reply docs/_getting-started/dev-mode.md]:
```suggestion
```

File: docs/_getting-started/dev-mode.md (hunk 3 above)
Comment (ChrisChinchilla) [reply docs/_getting-started/dev-mode.md]: I can't quite remember where this information came from, and I don't even understand it myself anymore now.

I think it was something to do with speeding up testing by effectively disabling features of the consensys engine. Is it best to cut this line or paraphrase what you say above?

Comment (holiman) [reply docs/_getting-started/dev-mode.md]: Write "Uses on-demand block generation, producing blocks when there are transactions waiting to be mined"

**Pull request merged**